internal/controller/factory/statefulset_test.go

```diff
@@ -129,6 +130,135 @@ var _ = Describe("CreateOrUpdateStatefulSet handler", func() {
 			By("Checking the extraArgs")
 			Expect(sts.Spec.Template.Spec.Containers[0].Command).To(Equal(generateEtcdCommand(etcdcluster)))
 
+			By("Checking the default startup probe")
+			Expect(sts.Spec.Template.Spec.Containers[0].StartupProbe).To(Equal(&v1.Probe{
+				ProbeHandler: v1.ProbeHandler{
+					HTTPGet: &v1.HTTPGetAction{
+						Path:   "/readyz?serializable=false",
+						Port:   intstr.FromInt32(2379),
+						Scheme: v1.URISchemeHTTP,
+					},
+				},
+				InitialDelaySeconds: 1,
+				TimeoutSeconds:      1,
+				PeriodSeconds:       5,
+				SuccessThreshold:    1,
+				FailureThreshold:    3,
+			}))
+
+			By("Checking the default readiness probe")
+			Expect(sts.Spec.Template.Spec.Containers[0].ReadinessProbe).To(Equal(&v1.Probe{
+				ProbeHandler: v1.ProbeHandler{
+					HTTPGet: &v1.HTTPGetAction{
+						Path:   "/readyz",
+						Port:   intstr.FromInt32(2379),
+						Scheme: v1.URISchemeHTTP,
+					},
+				},
+				InitialDelaySeconds: 5,
+				TimeoutSeconds:      1,
+				PeriodSeconds:       5,
+				SuccessThreshold:    1,
+				FailureThreshold:    3,
+			}))
+
+			By("Checking the default liveness probe")
+			Expect(sts.Spec.Template.Spec.Containers[0].LivenessProbe).To(Equal(&v1.Probe{
+				ProbeHandler: v1.ProbeHandler{
+					HTTPGet: &v1.HTTPGetAction{
+						Path:   "/livez",
+						Port:   intstr.FromInt32(2379),
+						Scheme: v1.URISchemeHTTP,
+					},
+				},
+				InitialDelaySeconds: 5,
+				TimeoutSeconds:      1,
+				PeriodSeconds:       5,
+				SuccessThreshold:    1,
+				FailureThreshold:    3,
+			}))
+
+			By("Deleting the statefulset")
+			Expect(k8sClient.Delete(ctx, sts)).To(Succeed())
+		})
+
+		It("should successfully override probes", func() {
+			By("Creating the statefulset")
+			etcdcluster := etcdcluster.DeepCopy()
+			etcdcluster.Spec.PodSpec = etcdaenixiov1alpha1.PodSpec{
+				LivenessProbe: &v1.Probe{
+					InitialDelaySeconds: 13,
+					PeriodSeconds:       11,
+				},
+				ReadinessProbe: &v1.Probe{
+					PeriodSeconds: 3,
+				},
+				StartupProbe: &v1.Probe{
+					PeriodSeconds: 7,
+					ProbeHandler: v1.ProbeHandler{
+						HTTPGet: &v1.HTTPGetAction{
+							Path: "/test",
+							Port: intstr.FromInt32(2389),
+						},
+					},
+				},
+			}
+
+			sts := &appsv1.StatefulSet{}
+			err := CreateOrUpdateStatefulSet(ctx, etcdcluster, k8sClient, k8sClient.Scheme())
+			Expect(err).NotTo(HaveOccurred())
+
+			err = k8sClient.Get(ctx, typeNamespacedName, sts)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Checking the updated startup probe")
```

Comment:
Please use `By` as follows:
```go
By("Checking the updated startup probe", func () {
// all code goes here
})
```

Reply:
Fixed my changes, but I see in other places the same pattern, maybe fix it in other pr